### PR TITLE
Remove `RuntimeApi` dependency on system parachain runtime code

### PR DESCRIPTION
**Pull request by seadanda**
The last issue blocking the removal of the Polkadot and Kusama system parachains from the repo in #1737 is the dependency on the runtime code through the RuntimeApi in `polkadot-parachain`.

This PR introduces two fake runtimes to satisfy the build requirements and changes the `new_partial` function to make it not be generic over the runtimes.
The reason for the second runtime is the different Aura keys used in Polkadot Asset Hub, as the impl for AuraApi depends on this type.
If this changes the `RuntimeApi` generic could be removed completely from all functions in `services.rs` and and generic type parameters in `services.rs` and specified as a concrete type to TFullClient`.

### CODE REVIEW

File: cumulus/polkadot-parachain/src/fake_runtime_api.rs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+			tx: <Block as BlockT>::Extrinsic,
+			block_hash: <Block as BlockT>::Hash,
+		) -> TransactionValidity {
+		unimplemented!()
+		}
+	}
+
+	impl sp_offchain::OffchainWorkerApi<Block> for Runtime {
+		fn offchain_worker(header: &<Block as BlockT>::Header) {
+		unimplemented!()
```
Comment (bkchr) [inline cumulus/polkadot-parachain/src/fake_runtime_api.rs:113]: FMT here and in multiple other places in the file.

Comment (seadanda) [reply cumulus/polkadot-parachain/src/fake_runtime_api.rs:113]: It's currently a WIP, just a few last missing impls being reported in the fake runtime which have insane type signatures 🙈.

File: cumulus/polkadot-parachain/Cargo.toml [hunk 2]
```diff
@@ -49,8 +49,10 @@ sp-runtime = { path = "../../substrate/primitives/runtime", default-features = f
 sp-io = { path = "../../substrate/primitives/io" }
 sp-core = { path = "../../substrate/primitives/core" }
 sp-session = { path = "../../substrate/primitives/session" }
+frame-try-runtime = { path = "../../substrate/frame/try-runtime", default-features = false, optional = true}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/Cargo.toml:52]:
```suggestion
frame-try-runtime = { path = "../../substrate/frame/try-runtime", optional = true }
```

File: cumulus/polkadot-parachain/Cargo.toml [hunk 3]
```diff
@@ -49,8 +49,10 @@ sp-runtime = { path = "../../substrate/primitives/runtime", default-features = f
...
 sc-consensus = { path = "../../substrate/client/consensus/common" }
 sp-tracing = { path = "../../substrate/primitives/tracing" }
+frame-support = { path = "../../substrate/frame/support", default-features = false}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/Cargo.toml:55]:
```suggestion
frame-support = { path = "../../substrate/frame/support" }
```

File: cumulus/polkadot-parachain/Cargo.toml [hunk 4]
```diff
@@ -63,12 +65,19 @@ sc-network-sync = { path = "../../substrate/client/network/sync" }
...
 sp-blockchain = { path = "../../substrate/primitives/blockchain" }
+sp-genesis-builder = { path = "../../substrate/primitives/genesis-builder", default-features = false }
 sp-block-builder = { path = "../../substrate/primitives/block-builder" }
 sp-keystore = { path = "../../substrate/primitives/keystore" }
 sc-chain-spec = { path = "../../substrate/client/chain-spec" }
 sc-rpc = { path = "../../substrate/client/rpc" }
+sp-version = { path = "../../substrate/primitives/version" }
 sc-tracing = { path = "../../substrate/client/tracing" }
 sp-offchain = { path = "../../substrate/primitives/offchain" }
+frame-system-rpc-runtime-api = { path = "../../substrate/frame/system/rpc/runtime-api", default-features = false}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/Cargo.toml:76]:
```suggestion
frame-system-rpc-runtime-api = { path = "../../substrate/frame/system/rpc/runtime-api" }
```

File: cumulus/polkadot-parachain/Cargo.toml [hunk 5]
```diff
@@ -63,12 +65,19 @@ sc-network-sync = { path = "../../substrate/client/network/sync" }
...
+pallet-transaction-payment = { path = "../../substrate/frame/transaction-payment", default-features = false}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/Cargo.toml:77]:
```suggestion
pallet-transaction-payment = { path = "../../substrate/frame/transaction-payment" }
```

File: cumulus/polkadot-parachain/Cargo.toml [hunk 6]
```diff
@@ -63,12 +65,19 @@ sc-network-sync = { path = "../../substrate/client/network/sync" }
...
+pallet-transaction-payment-rpc-runtime-api = { path = "../../substrate/frame/transaction-payment/rpc/runtime-api", default-features = false}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/Cargo.toml:78]:
```suggestion
pallet-transaction-payment-rpc-runtime-api = { path = "../../substrate/frame/transaction-payment/rpc/runtime-api" }
```

File: cumulus/polkadot-parachain/Cargo.toml [hunk 7]
```diff
@@ -63,12 +65,19 @@ sc-network-sync = { path = "../../substrate/client/network/sync" }
...
+sp-std = { path = "../../substrate/primitives/std", default-features = false}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/Cargo.toml:79]:
```suggestion
sp-std = { path = "../../substrate/primitives/std" }
```

File: cumulus/polkadot-parachain/Cargo.toml [hunk 8]
```diff
@@ -63,12 +65,19 @@ sc-network-sync = { path = "../../substrate/client/network/sync" }
...
+sp-inherents = { path = "../../substrate/primitives/inherents", default-features = false}
```
Comment (bkchr) [inline cumulus/polkadot-parachain/Cargo.toml:80]:
```suggestion
sp-inherents = { path = "../../substrate/primitives/inherents" }
```

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
And CI is also failing.

**Review by franciscoaguirre — APPROVED** [review]

**Pull request merged**